src/lib/storage/proposal/partition_creator.rb

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "fileutils"
+require "storage/planned_volumes_collection"
+require "storage/disk_size"
+require "storage/proposal/refined_devicegraph"
+require "storage/proposal/exceptions"
+
+module Yast
+  module Storage
+    module Proposal
+      using RefinedDevicegraph
+
+      # Class to create partitions in the free space detected or freed by the
+      # SpaceMaker.
+      class PartitionCreator
+        include Yast::Logger
+
+        attr_accessor :settings
+
+        VOLUME_GROUP_SYSTEM = "system"
+        FIRST_LOGICAL_PARTITION_NUMBER = 5 # Number of the first logical partition (/dev/sdx5)
+
+        # Initialize.
+        #
+        # @param original_graph [::Storage::Devicegraph] initial devicegraph
```

Comment:
"Initial devicegraph" is misleading. I wrote that code, and I couldn't make sense of this.

This is nothing like an "original graph", it's the device graph this object works on. It is changed by this class, and the changes are visible to the outside, so this is not anything like an "initial" or "original" device graph -- which would imply that this class copies it or whatever. This is not true.

This is the device graph to work on, as the original documentation said. Nothing more, nothing less. Let's not confuse people who read this.